src/main/java/br/edu/utfpr/reclamaguarapuava/controller/NeighborhoodController.java

```diff
@@ -1,13 +1,112 @@
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import br.edu.utfpr.reclamaguarapuava.model.Neighborhood;
+import br.edu.utfpr.reclamaguarapuava.model.dto.NeighborhoodDTO;
+import br.edu.utfpr.reclamaguarapuava.model.service.NeighborhoodService;
+import br.edu.utfpr.reclamaguarapuava.util.Response;
+import br.edu.utfpr.reclamaguarapuava.util.erros.InvalidParamsException;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
 @RestController
-@RequestMapping("/api/v1/bairros")
-@Api(value = "Neighborhood Controller")
+@RequestMapping("/api/v1/admin/bairros")
+@Api(value = "Neighborhood", description = "Url para manipular ou buscar bairros")
 public class NeighborhoodController {
 
+	private final NeighborhoodService neighborhoodService;
+
+	private static final Logger log = LoggerFactory.getLogger(NeighborhoodController.class);
+
+	@Autowired
+	public NeighborhoodController(NeighborhoodService neighborhoodService) {
+		this.neighborhoodService = neighborhoodService;
+	}
+
+	@GetMapping
+	@ApiOperation(value = "Retorna todas os bairros", notes = "A lista retornada é paginada")
+	@ApiResponses(value = { @ApiResponse(code = 200, message = "Quando bem sucedida para todos os bairros"),
+			@ApiResponse(code = 403, message = "Acesso negado"),
+			@ApiResponse(code = 500, message = "Quando a requisição causou um error interno no servidor"), })
+	public ResponseEntity<Page<Neighborhood>> getAll(Pageable pageable) {
+		log.debug("Request GET to '/api/v1/admin/bairros' in process");
+		return new ResponseEntity<>(neighborhoodService.findAll(pageable), HttpStatus.OK);
+	}
+
+	@GetMapping(value = "/{id}")
+	@ApiOperation(value = "Retorna um único bairro")
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = "Quando bem sucedida retorna os detalhes de um bairro", response = Neighborhood.class),
+			@ApiResponse(code = 403, message = "Acesso negado"),
+			@ApiResponse(code = 404, message = "Bairro não existe na base de dados"),
+			@ApiResponse(code = 500, message = "Quando a requisição causou um error interno no servidor"), })
+	public ResponseEntity<Response<Neighborhood>> getById(
+			@ApiParam(name = "id", value = "Númeral do tipo long que corresponde ao id do bairro a qual requer detalhes") @PathVariable("id") Long id) {
+		log.debug("Request GET to '/api/v1/admin/bairros/" + id + "' in process");
+		return new ResponseEntity<>(neighborhoodService.findById(id), HttpStatus.OK);
```

Comment:
Verificar se  consegue encontrar o objeto pelo id, cado contrário, mandar uma mensagem de erro no Response. Neste caso, enviar um badRequest()

Reply:
{"title":"Resource not found","statusCode":404,"timestamp":1543938669253,"exception":"EntityNotFoundException","error":"neighborhood not found by id: 500"}